test/langtools/tools/javac/reflect/MethodReferenceTest.java

```diff
@@ -168,7 +168,7 @@ void test6() {
     @IR("""
             func @"test7" (%0 : MethodReferenceTest)void -> {
                 %1 : java.util.function.Supplier<.<MethodReferenceTest, MethodReferenceTest$A<java.lang.String>>> = lambda ().<MethodReferenceTest, MethodReferenceTest$A<java.lang.String>> -> {
-                    %2 : .<MethodReferenceTest, MethodReferenceTest$A<java.lang.String>> = new %0 @"func<.<MethodReferenceTest, MethodReferenceTest$A>>";
+                    %2 : .<MethodReferenceTest, MethodReferenceTest$A<java.lang.String>> = new %0 @"func<.<MethodReferenceTest, MethodReferenceTest$A>, MethodReferenceTest>";
```

Comment:
Note that for inner classes we used to generate incorrect constructor types (as the enclosing this type was missing)